Improve pulseaudio backend

pedrocr opened the pull request:
A few improvements for pulseaudio:

- Fix some uses of deprecated API
- Add a way to set a volume limit
- Expose mute/unmute in the CLI

pedrocr committed:
Fix deprecated uses of pulseaudio bindings

kennylevinsen commented on src/audio/pulseaudio.rs:
```diff
@@ -362,10 +363,15 @@ impl PulseAudioSoundDevice {
         };
 
+        // Since we're adjusting the volume unmute
+        self.set_muted(false)?;
```
I do not think this is the correct thing to do: If the user wishes to unmute on change, they could simply call set_muted(false) themselves. With this, you cannot avoid it.

I personally adjust my volume before I unmute to not accidentally burst my eardrums.

pedrocr replied:
Yeah, that's the other option and pacmd does it like that. I'll need to expose the mute/unmute as well so I can chain two commands in one bindsym like I do with pacmd.

kennylevinsen reviewed (changes requested):
Thanks for the API upgrade. I do have one comment, but otherwise good.

pedrocr commented on src/audio/pulseaudio.rs:
```diff
@@ -15,7 +14,9 @@ use libpulse_binding::context::{
-use libpulse_binding::volume::{ChannelVolumes, VOLUME_MAX, VOLUME_NORM};
+use libpulse_binding::volume::{ChannelVolumes, Volume, VOLUME_NORM};
+
+static MAX_VOLUME: Volume = Volume(131072); // 2^16 * 2 = 200%
```
Any idea if there's a more adequate value? I was surprised to see pulse audio allow the volume to go up to 2000% and more, but now that I check pacmd doesn't have a limit either.

kennylevinsen replied:
Honestly, I think any limit we'd be setting is arbitrary, so I'm tempted to say that there shouldn't be a limit.

pedrocr replied:
Then maybe there should be an optional argument to add to set the cap. Something like:

```
platformctl audio -- +0.10 1.5
```

To increase volume by 10% up to 150% max.

kennylevinsen replied:
That sounds reasonable.

pedrocr committed:
Expose both volume and mute/unmute

CLI is now:

  platformctl audio volume <increment>
  platformctl audio mute <on|off|toggle>

pedrocr committed:
Add a maximum value to the volume

The CLI now becomes:

  platformctl audio volume <step> [max]

with the optional maximum value being used to limit the volume.

kennylevinsen reviewed (approved):
Awesome, thanks!
